src/server/editorServices.ts

```diff
@@ -17,6 +17,58 @@ namespace ts.server {
         (event: ProjectServiceEvent): void;
     }
 
+    const indentStyle = createMap({
+        "none": IndentStyle.None,
+        "block": IndentStyle.Block,
+        "smart": IndentStyle.Smart
+    });
+
+    export function convertFormatOptions(protocolOptions: protocol.FormatCodeSettings): FormatCodeSettings {
+        if (typeof protocolOptions.indentStyle === "string") {
+            protocolOptions.indentStyle = indentStyle[protocolOptions.indentStyle.toLowerCase()];
+            Debug.assert(protocolOptions.indentStyle !== undefined);
+        }
+        return <any>protocolOptions;
+    }
+
+    export function convertCompilerOptions(protocolOptions: protocol.ExternalProjectCompilerOptions): CompilerOptions & protocol.CompileOnSaveMixin {
+        protocolOptions.target = convertCompilerOptionEnum(targetCommandLineOption, protocolOptions.target);

```

Comment:
can we do this in a more generic way? we have the types for the options as a whole, so we can just go through them once and build a map of things we want to convert, then do this on the fly?